examples/vault-ec2-auth/main.tf

```diff
@@ -0,0 +1,210 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# DEPLOY A VAULT SERVER CLUSTER AND A CONSUL SERVER CLUSTER IN AWS
+# This is an example of how to launch a vault cluster and then authenticate an instance to the cluster
+# ---------------------------------------------------------------------------------------------------------------------
+
+terraform {
+  required_version = ">= 0.11.7"
+}
+
+resource "aws_instance" "example_auth_to_vault" {
+  ami           = "${var.ami_id}"
+  instance_type = "t2.micro"
+  subnet_id     = "${data.aws_subnet_ids.default.ids[0]}"
+  key_name      = "${var.ssh_key_name}"
+
+  # Uses the same security group as the vault cluster
+  # Plus an additional one that opens the port to our simple web server
+  security_groups = [
+    "${module.vault_cluster.security_group_id}",
+    "${aws_security_group.auth_instance.id}",
+  ]
+
+  user_data            = "${data.template_file.user_data_auth_client.rendered}"
+  iam_instance_profile = "${aws_iam_instance_profile.example_instance_profile.name}"
+
+  tags {
+    Name = "${var.auth_server_name}"
+  }
+}
+
+# Using the same role as vault because it has consul_iam_policies_servers,
+# Which allows DescribeTags and enables this instance to connect to find and reach consul server
+# access the DNS registry for the vault server
```

Comment:
The permissions for this role come from this module, which you could also use directly: https://github.com/hashicorp/terraform-aws-consul/tree/master/modules/consul-iam-policies

Reply:
Yes. I mentioned this in the doc I wrote. Gonna push it today